src/hypothesis/internal/conjecture/utils.py

```diff
@@ -68,7 +68,14 @@ def byte_distribution(random, n):
         else:
             probe = upper - v
         return int_to_bytes(probe, n)
-    probe = int_from_bytes(data.draw_bytes(nbytes, byte_distribution)) & mask
+
+    probe = gap + 1
+
+    while probe > gap:  # pragma: no branch
+        probe = int_from_bytes(
+            data.draw_bytes(nbytes, byte_distribution)
+        ) & mask
+
     if probe <= gap:
```

Comment:
Sorry, I misdiagnosed the coverage issue before. The actual problem is that this condition is now always satisfied because the previous bit loops until it is, so the if is now redundant.